detectable/src/main/java/com/synopsys/integration/detectable/detectables/clang/linux/LinuxDistro.java

```diff
@@ -17,9 +19,29 @@ public class LinuxDistro {
 
     public Optional<String> extractLinuxDistroNameFromEtcDir() {
         File etcDir = new File("/etc");
+
+        if (isOracleLinux(etcDir)) {
+            logger.trace("Target image Linux distro is an Oracle Linux");
+            return Optional.of("oracle_linux");
+        }
+
         return extractLinuxDistroNameFromEtcDir(etcDir);
     }
 
+    public boolean isOracleLinux(File etcDirFile) {
+        for (File etcFile : etcDirFile.listFiles()) {
+            if ("oracle-release".equals(etcFile.getName())) {
+                logger.trace("Found oracle-release");
+                return true;
+            }
+        }
+
+        File releaseFile = new File(etcDirFile, "os-release");
+        Optional<String> distroName = getLinuxDistroNameFromStandardReleaseFile(releaseFile);
```

Comment:
Question here: if "os-release" file does not already exist, won't this create a new file named "os-release" in the etc directory? 
What would `getLinuxDistroNameFromStandardReleaseFile` return if this is a new file that was created?
Or should the function just return false if "os-release" is not found?

Reply:
It won't create new file if it is exists. We consider the "os-release" file will be exists.

Reply:
Will it create a new file if "os-release" does not exist?